src/main/java/homer/model/temperaturechangers/AirConditioning.java

```diff
@@ -0,0 +1,37 @@
+package homer.model.temperaturechangers;
+
+import java.time.Duration;
+
+import homer.common.Temperature;
+import homer.common.TemperatureFactory;
+import homer.model.environment.Environment;
+
+/**
+ * Models air conditioning.
+ */
+public final class AirConditioning extends AbstractTemperatureChanger {
+
+
+    /**
+     * 
+     * @param minIntensity
+     * @param maxIntensity
+     * @param currentIntensity
+     * @param environment
+     * @param minTemperature
+     * @param maxTemperature
+     */
+    public AirConditioning(final double minIntensity, final double maxIntensity, final double currentIntensity, 
+        final Environment environment, final Temperature minTemperature, final Temperature maxTemperature) {
```

Comment:
The constructors feels a bit bloated, maybe min-max ranges can be coupled with a pair or some other data structure.
currentIntensity might be unnecessary, could just be set to the minimum

Reply:
I agree the parameters are way too many, i think setting `currentIntensity` to the minimum might be worth it

Reply:
I think i will switch to a builder-based construction and declutter the constructor by passing sane defaults for `minTemperature` and `MaxTemperature`